model/riscv_insts_vext_utils.sail

```diff
@@ -0,0 +1,594 @@
+  mask_helper       : vector('n, dec, bool) = undefined;
+  result            : vector('n, dec, bool) = undefined;
+
+  /* Determine the actual number of elements when lmul < 1 */
+  let real_num_elem = if lmul >= 1.0 then num_elem else floor(lmul * to_real(num_elem));
+  assert(num_elem >= real_num_elem);
+
+  foreach (i from 0 to (num_elem - 1)) {
+    if i < start_element then {
+      /* Prestart elements defined by vstart */
+      result[i] = vd_val[i];
+      mask_helper[i] = false
+    } else if i > end_element then {
+      /* Tail elements defined by vl */
+      /* Mask tail is always agnostic */
+      result[i] = vd_val[i];
+      mask_helper[i] = false
+    } else if i >= real_num_elem then {
+      /* Tail elements defined by lmul < 1 */
+      /* Mask tail is always agnostic */
+      result[i] = vd_val[i];
+      mask_helper[i] = false
+    } else {
+      /* Active body elements */
+      mask_helper[i] = true
+    }
+  };
+
+  (result, mask_helper)
+}
+
+/* Mask handling for cmp functions that use masks as output */
+val init_masked_result_cmp : forall 'n 'm, 8 <= 'm <= 128. (int('n), int('m), real, vector('n, dec, bool), vector('n, dec, bool)) -> (vector('n, dec, bool), vector('n, dec, bool)) effect {escape, rreg, undef, wreg}
+function init_masked_result_cmp(num_elem, vsew_bits, lmul, vd_val, vm_val) = {
+  let start_element : int                       = get_start_element();
+  let end_element   : int                       = get_end_element();
+  let mask_ag       : agtype                    = get_vtype_vma();
+  mask_helper       : vector('n, dec, bool)     = undefined;
+  result            : vector('n, dec, bool)     = undefined;
+
+  /* Determine the actual number of elements when lmul < 1 */
+  let real_num_elem = if lmul >= 1.0 then num_elem else floor(lmul * to_real(num_elem));
+  assert(num_elem >= real_num_elem);
+
+  foreach (i from 0 to (num_elem - 1)) {
+    if i < start_element then {
+      /* Prestart elements defined by vstart */
+      result[i] = vd_val[i];
+      mask_helper[i] = false
+    } else if i > end_element then {
+      /* Tail elements defined by vl */
+      /* Mask tail is always agnostic */
+      result[i] = vd_val[i];
+      mask_helper[i] = false
+    } else if i >= real_num_elem then {
+      /* Tail elements defined by lmul < 1 */
+      /* Mask tail is always agnostic */
+      result[i] = vd_val[i];
+      mask_helper[i] = false
+    } else if vm_val[i] == false then {
+      /* Inactive body elements defined by vm */
+      if mask_ag == UNDISTURBED then {
+        result[i] = vd_val[i]
+      } else if mask_ag == AGNOSTIC then {
+	      result[i] = vd_val[i]
+      };
+      mask_helper[i] = false
+    } else {
+      /* Active body elements */
+      mask_helper[i] = true
+    }
+  };
+
+  (result, mask_helper)
+}
+
+/* Misc number functions */
+val is_zeros : forall 'n, 'n >= 0. bits('n) -> bool
+function is_zeros(bitval) = {
```

Comment:
Not sure why you need a function for these rather than just `== zeros/ones()` everywhere?

Reply:
These two functions are removed